part1/stages/Check-GPU

```diff
@@ -56,4 +56,102 @@ if [ ${GPU_OK} -ne 0 ] ; then
     echo "Ignoring unrecognised graphics device.">&2
 fi
 
+get_vga_pci_info()
+{
+    lspci -m | grep "VGA"
+}
+
+get_pci_id()
+{
+    echo $1 | awk '{print $1}'
+}
+
+get_pci_bus()
+{
+    echo $1 | cut -d: -f 1
+}
+
+get_onboard_pci_id()
+{
+    OLD_IFS=$IFS
+    IFS="
+"
+    for vga_dev in $(get_vga_pci_info); do
+        vga_id="$(get_pci_id $vga_dev)"
+        if [ "$(get_pci_bus $vga_id)" -eq "00" ]; then
+            echo "$vga_id"
+            break
+        fi
+    done
+    IFS=${OLD_IFS}
+}
```

Comment:
`lspci -s 00: -d ::0300` should give you the VGA compatible controller on the host bridge (excluding external card potentially setup as Primary) and do the lifting for you :).